filament/backend/src/opengl/PlatformEGL.cpp

```diff
@@ -75,21 +75,22 @@ static void clearGlError() noexcept {
 
 // ---------------------------------------------------------------------------------------------
 
-PlatformEGL::PlatformEGL(EGLDisplay display) noexcept : 
-        mEGLDisplay(display) {
+PlatformEGL::PlatformEGL(EGLDisplay display) noexcept :
+        mEGLDisplay(display), mIsEGLDisplayExternal(display != EGL_NO_DISPLAY) {
 }
 
 Driver* PlatformEGL::createDriver(void* sharedContext) noexcept {
-    if (mEGLDisplay == EGL_NO_DISPLAY) {
+    if (!mIsEGLDisplayExternal) {
         mEGLDisplay = eglGetDisplay(EGL_DEFAULT_DISPLAY);
         assert_invariant(mEGLDisplay != EGL_NO_DISPLAY);
-    }
 
-    EGLint major, minor;
-    EGLBoolean initialized = eglInitialize(mEGLDisplay, &major, &minor);
-    if (UTILS_UNLIKELY(!initialized)) {
-        slog.e << "eglInitialize failed" << io::endl;
-        return nullptr;
+        EGLint major, minor;
+        EGLBoolean initialized = eglInitialize(mEGLDisplay, &major, &minor);
+        if (UTILS_UNLIKELY(!initialized)) {
+            slog.e << "eglInitialize failed" << io::endl;
+            return nullptr;
+        }
+
```

Comment:
The whitespace changes are needed, because this block of code was moved into the if to avoid reinitialization of the EGLDisplay, except the new line, but I would rather not recompile and stage filament libs because of one line.